pubsub/google/cloud/pubsub_v1/subscriber/_protocol/dispatcher.py

```diff
@@ -34,6 +37,18 @@
 """The maximum amount of time in seconds to wait for additional request items
 before processing the next batch of requests."""
 
+_ACK_IDS_BATCH_SIZE = 3000
+"""The maximum number of ACK IDs to send in a single StreamingPullRequest.
+
+The backend imposes a maximum request size limit of 524288 bytes (512 KiB) per
+acknowledge / modifyAckDeadline request. ACK IDs have a maximum size of 164
+bytes, thus we cannot send more than o 524288/164 ~= 3197 ACK IDs in a single
+StreamingPullRequest message.
```

Comment:
Picked the max ACK ID size from [here](https://github.com/googleapis/nodejs-pubsub/pull/65/files#diff-3d29c4447546c72118ed5d5cbf38ab8bR37) (a similar fix in the Node.js client).

Reply:
We don't publish a set size for an ack id. It may change over time, though we rarely do so in practice. 2500 is fine for now.

Reply:
Indeed. We discussed this at one of the meetings, and it was said that any changes to ACK ID sizes are both rare and minimal in terms of space. Since 2500 gives us a safety buffer of approximately ~20%, it is enough in practice.